src/mono/mono/mini/memory-access.c

```diff
@@ -130,6 +130,12 @@ mini_emit_memcpy (MonoCompile *cfg, int destreg, int doffset, int srcreg, int so
 	}
 
 	g_assert (align > 0);
+	
+#ifndef NO_UNALIGNED_ACCESS	
+	// Unaligned access is not a problem on this platform. Use the longest
+	// possible vector.
+	goto copy_16;
```

Comment:
Note that the cpu allowing unaligned access does not mean that its going to be fast.

Reply:
What I'm trying to do is optimize memcpying types like `Matrix4x4` which have proper alignment, but do not register as SIMD types. Other than checking class name against a known list, do you see a way of detecting those?

Reply:
The function has an align argument which is supposed to be the guaranteed alignment of src/dest.